Makefile

```diff
@@ -1,172 +1,383 @@
+	@$(OK) Performing Docker scan report
+
+.PHONY: docker-lint
+docker-lint: ## to lint the Dockerfile
+	@$(INFO) Dockerfile linting...
+	$(AT)$(DOCKER) run -i ${DOCKER_OPTS} \
+	${DOCKER_IMAGE_DOCKERLINT} \
+	< ${DOCKER_FILE} || ${FAIL}
+	@$(OK) Dockerfile linting
+
+.PHONY: docker-login
+docker-login: ## to login to a container registry
+	@$(INFO) Dockerd login to container registry ${DOCKER_REGISTRY}...
+	$(AT) echo "${DOCKER_PASSWORD}" | $(DOCKER) login --password-stdin -u ${DOCKER_USER} $(DOCKER_REGISTRY) || ${FAIL}
+	@$(OK) Dockerd login to container registry ${DOCKER_REGISTRY}...
+
+go-build: $(GO_BUILD_PLATFORMS_ARTIFACTS) ## to build binaries
+
+.PHONY: go-build
+go-build/%:
+	@$(INFO) go build $*...
+	$(AT)target="$*"; \
+	command="${APP_NAME}"; \
+	platform_ext="$${target#$$command-*}"; \
+	platform="$${platform_ext%.*}"; \
+	export GOOS="$${platform%%-*}"; \
+	export GOARCH="$${platform#*-}"; \
+	echo export GOOS=$${GOOS}; \
+	echo export GOARCH=$${GOARCH}; \
+	CGO_ENABLED=0 \
+	$(GO) build ${GO_BUILD_OPTS} \
+	-ldflags '${GO_LDFLAGS}' \
+	-o ${GO_OUT_BIN_DIR}/$* \
+	${CONFIG_APP_CODE} || ${FAIL}
+	@$(OK) go build $*
+
+.PHONY: go-build-docker
+go-build-docker: # to build binaries under a controlled docker dedicated go container using DOCKER_IMAGE_GO
+	@$(INFO) go build docker
+	$(AT)$(DOCKER) run  \
+	-v $(PWD):/app -w /app \
+	$(DOCKER_IMAGE_GO) \
+	/bin/sh -c \
+	"cd /app && \
+	make go-build"  || ${FAIL}
+	@$(OK) go build docker
+
+.PHONY: go-run
+go-run: ## to run locally for development
+	@$(INFO) running locally...
+	$(AT)$(GO) run ${GO_BUILD_OPTS} ${CONFIG_APP_CODE} || ${FAIL}
+	@$(OK) running locally
+
+.PHONY: go-test
+go-test: ## to run tests
+	@$(INFO) testing...
+	$(AT)$(DOCKER) run ${DOCKER_OPTS} \
+	-v $(PWD):/app -w /app \
+	-e GOCACHE="/tmp" \
+	$(DOCKER_IMAGE_GO) \
+	/bin/sh -c \
+	"cd /app && \
+	go test ${GO_TEST_OPTS} ./... " || ${FAIL}
+	@$(OK) testing
+
+.PHONY: go-mod-check
+go-mod-check: ## to check go mod files consistency
+	@$(INFO) Checking go mod files consistency...
+	$(AT)$(GO) mod tidy
+	$(AT)git --no-pager diff --exit-code go.mod go.sum || \
+	(${WARN} Please run "go mod tidy" and commit the changes in go.mod and go.sum. && ${FAIL} ; exit 128 )
+	@$(OK) Checking go mod files consistency
+
+.PHONY: go-update-dependencies
+go-update-dependencies: ## to update go dependencies (vendor)
+	@$(INFO) updating go dependencies...
+	$(AT)$(GO) get -u ./... && \
+	$(AT)$(GO) mod vendor && \
+	$(AT)$(GO) mod tidy || ${FAIL}
+	@$(OK) updating go dependencies
+
+.PHONY: go-lint
+go-lint: ## to lint go code
+	@$(INFO) App linting...
+	$(AT)GOCACHE="/tmp" $(DOCKER) run ${DOCKER_OPTS} \
+	-v $(PWD):/app -w /app \
+	-e GOCACHE="/tmp" \
+	-e GOLANGCI_LINT_CACHE="/tmp" \
+	${DOCKER_IMAGE_GOLINT} \
+	golangci-lint run ./... || ${FAIL}
+	@$(OK) App linting
+
+.PHONY: go-fmt
+go-fmt: ## to perform formatting
+	@$(INFO) App code formatting...
+	$(AT)$(GO) fmt ./... || ${FAIL}
+	@$(OK) App code formatting...
```

Comment:
This isn't really needed. Golangci does this automatically. Can be removed.

Reply:
🤝

Reply:
Rebased and pushed fix.